### PR TITLE
Upgrade to sbt 1.2.3, properly report problems to sbt

### PR DESCRIPTION
**Pull request by smarter**

### CODE REVIEW

File: project/Build.scala [hunk 1]
```diff
@@ -928,6 +928,7 @@ object Build {
   // sbt plugin to use Dotty in your own build, see
   // https://github.com/lampepfl/dotty-example-project for usage.
   lazy val `sbt-dotty` = project.in(file("sbt-dotty")).
+    enablePlugins(ScriptedPlugin).
```
Comment (allanrenucci) [inline project/Build.scala]: Should we use the newer [SbtPlugin](https://developer.lightbend.com/blog/2018-07-02-sbt-1-2-0/#sbtplugin-for-plugin-development)?

**Comment by allanrenucci** [comment]
We should experiment with https://github.com/sbt/sbt/pull/3891. It introduces two new options to the scripted tests:
- scriptedBatchExecution (Boolean): Enables or disables batch execution for scripted.
- scriptedParallelInstances (Int): Configures the number of scripted instances for parallel testing, only used in batch mode.

**Comment by smarter** [comment]
We're getting a ClassCastException in the scripted test but it's probably harmless: https://github.com/sbt/sbt/issues/4361

**Comment by smarter** [comment]
> scriptedBatchExecution (Boolean): Enables or disables batch execution for scripted.

> sbt.internal.scripted.TestException: {line -1} Reload for scripted batch execution failed.

Looks like a no go.

**Comment by smarter** [comment]
Also before that there is:
> [error] Error: Could not find or load main class sbt.RunFromSourceMain

**Comment by allanrenucci** [comment]
Too bad. No speedup for the scripted tests then...

**Pull request merged**